cachet-tools/update_cachet.py

```diff
@@ -0,0 +1,114 @@
+INSIGHTS_PASSWORD = os.getenv("INSIGHTS_PASSWORD")
+INSIGHTS_SERVER   = os.getenv("INSIGHTS_SERVER")
+API_URL           = f"https://{INSIGHTS_SERVER}/api"
+
+# format the logger message
+logging.basicConfig(level='INFO', format='%(asctime)s %(message)s', datefmt='%m/%d/%Y %H:%M:%S %p: ')
+logger = logging.getLogger(__name__)
+
+# session used by every request, only URLs and args will change
+session  = requests.Session()
+services = Services()
+
+def update_all_services():
+    logger.info("Starting services status check ...")
+
+    # session = requests.Session()
+    session.headers.update(HEADERS)
+
+    try:
+        response = session.get(CACHET_URL + "/groups", verify=False)
+        groups   = response.json()['data']
+        logger.info("Number of groups found: " + str(len(groups)))
+
+        for group in groups:  # for somereason not using groups returns 20 components but should return 23
+            components = group['enabled_components']
+            logger.info("Starting to check each of " + str(len(components)) + " components in group \"" + group['name'] + "\"")
+            for component in components:
+                # assuming the component in the parent group is accessible because it showed up.
+                svc_status = get_service_status(component['name'])
+                update_svc = set_svc_status(component['id'], svc_status)
+                if update_svc:
+                    logger.info("Successfully updated status of \"" + component['name'] + "\" service")
+                else:
+                    logger.error("Failed to updated status of \"" + component['name'] + "\" service")
+                time.sleep(2) 
+            # end of for components loop
+        # end of for groups loop
+    except Exception as ex:
+        logger.exception(ex)
+    # end of try/except block
+
+    logger.info("Done updating all services!!!")
+# end of def update_all_services
+
+def get_service_status(name):
+    result = True # set to False if exception thrown
+    uri    = services.getServiceUriByLabel(name)
+    url    = API_URL + "/" + uri
+    try:
+        response = session.get(url, auth=(INSIGHTS_USERNAME, INSIGHTS_PASSWORD), verify=False, )
+        logger.info(str(response.status_code) + " received from " + url)
+        result = response.status_code
+    except requests.exceptions.RequestException as re:
+        logger.info("Error reading service \"" + name + "\" from Insights")
+        logger.exception(re)
+        result = False
+    return result
+# end of def get_service_status
+
+def set_svc_status(id, status):
+    result = True # set to False by the exception handler
+    url    = CACHET_URL + "/" + str(id)
+
+    try:
+        # change status to "Unknown" temporarily before setting the actual status.
+        # reason: time-stamp is not updated in cachet, when before and after status is the same.
+        ret = session.put(url, data={"status": 0}, verify=False)
+        ret = session.get(url, verify=False)
```

Comment:
Do you need this `get`?  You don’t do anything with the result.

Reply:
If the status remains the same before and after the check, even updating the status does not update the time stamp and the WeUI shows the update time since the last status change.  To ensure the timestamp update, I use this temporary change.  Thanks to Cachet API implementation.